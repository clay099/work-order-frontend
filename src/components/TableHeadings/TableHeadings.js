import React from "react";
import { TableHead, TableCell, TableRow } from "@material-ui/core/";

const TableHeadings = ({ headings }) => {
	return (
		<TableHead>
			<TableRow>
				{headings.map((heading) => (
					<TableCell>{heading}</TableCell>
				))}
			</TableRow>
		</TableHead>
	);
};

export default TableHeadings;
