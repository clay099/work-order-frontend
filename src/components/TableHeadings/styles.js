import { makeStyles } from "@material-ui/core/styles";
import { tableHead } from "../../styles/styleObjects";

const useStyles = makeStyles((theme) => {
	return tableHead(theme);
});

export default useStyles;
