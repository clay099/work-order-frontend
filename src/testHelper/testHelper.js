// testHelper.js
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { createStore } from "redux";
import { Provider } from "react-redux";
// Import your own reducer
import reducer from "../reducers/rootReducer";

function render(
	ui,
	{
		initialState,
		store = createStore(reducer, initialState),
		...renderOptions
	} = {}
) {
	function Wrapper({ children }) {
		return <Provider store={store}>{children}</Provider>;
	}
	return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const tokenState = { token: "token" };
const userTypeUserState = { userType: "user" };
const userTypeTradesmenState = { userType: "user" };
const idState = { id: 1234 };
const userDetailsState = { userDetails: "userDetails" };
const handleSubmitState = { handleSubmit: jest.fn() };
const handleChangeState = { handleChange: jest.fn() };
const disabledState = { disabled: false };
const disabledEmailState = { disabledEmail: false };
const buttonTextState = { buttonText: "buttonText" };
const formDataState = {
	formData: {
		id: 1,
		firstName: "firstName",
		lastName: "lastName",
		email: "email",
		password: "password",
		newPassword: "newPassword",
		confirmNewPassword: "confirmNewPassword",
	},
};

// re-export everything
export * from "@testing-library/react";
// override render method
export {
	render,
	tokenState,
	userTypeUserState,
	userTypeTradesmenState,
	idState,
	userDetailsState,
	handleSubmitState,
	handleChangeState,
	disabledState,
	disabledEmailState,
	buttonTextState,
	formDataState,
};
