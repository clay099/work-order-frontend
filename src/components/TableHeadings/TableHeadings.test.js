import React from "react";
import TableHeadings from "./TableHeadings";
import { render } from "../../testHelper/testHelper";
console.error = jest.fn();

describe("TableHeadings Component", () => {
	// smoke test
	test("renders without crashing", () => {
		render(<TableHeadings headings={["test"]} />);
	});

	// snapshot test
	test("matches snapshot", () => {
		const { asFragment } = render(<TableHeadings headings={["test"]} />);
		expect(asFragment).toMatchSnapshot();
	});

	test("value to be in the document", () => {
		const { getByText } = render(<TableHeadings headings={["test"]} />);
		const value = getByText("test");
		expect(value).toBeInTheDocument();
	});
});
