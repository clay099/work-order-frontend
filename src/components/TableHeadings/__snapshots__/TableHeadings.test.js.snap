// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeadings Component matches snapshot 1`] = `[Function]`;
